### DISCUSSION TITLE
Ant Media Server cannot start more than 8 stream sources

### DISCUSSION THREAD

**Discussion by JK100001** [ant-media/Ant-Media-Server#5068] [Q&A] [answered]
Hello everyone! I'm using the Ant Media Server to stream 20 rtsp threads to website. When i deployed antmedia by docker container and added stream sources, i could see all 20 streams via https://antmedia/WebRTCApp/play.thml&name=ID for each rtsp thread, everything worked well.
But recently my hardware server on which there was a virtual machine antmedia was crashed into reboot. After turning on, half of my stream sources has status "preparing" but nothing is going on.

In console logs i have messages for each rtsp thread:

2023-05-03 15:40:38,280 [Thread-12991] ERROR i.a.streamsource.StreamFetcher - cannot open stream: rtsp://IP:555/ID/ with error:: Invalid data found when processing input
2023-05-03 15:40:38,280 [Thread-12993] INFO  i.a.streamsource.StreamFetcher - write all buffered packets for stream: ID
2023-05-03 15:40:38,281 [Thread-12991] ERROR i.a.streamsource.StreamFetcher - Prepare for opening the rtsp://IP:555/ID/ has failed

Its strange because before turning off my server worked well. 
Now only 8 streams out of 20 are working. Moreover, when you restart antmedia, streams that did not work can start working, but the total number of working streams is always 8.

I ask you to help. I can't figure out what could be the problem.

**Comment by tremzo** [comment]
Please verify each stream is definitely working by using VLC player or similar and then can you try to turn them all off, restart Ant Media Server and then turn each stream on one by one and see if you can successfully turn on all 20 streams?

**Reply by JK100001** [reply]
I did this, in VLC stream rtsp played well, also I tried restarting ant media but I can't run more than 8 stream sources.  Only the first 8 threads were successfully launched, which I enabled in the first place

**Reply by mekya** [reply]
Hey @timantmedia,
Do you have any update?

**Comment by tremzo** [comment]
@JK100001 I wasn't able to reproduce the stream source limit of 8. I have AMS running locally in a docker and was able to pull 11 rtsp streams without issue. 

Please can you confirm your Ant Media Server version? It would also be very helpful if you could provide your log and error files so we can see if anything indicates the root cause. 

What are the specifications of your hardware server? What caused the crash? Did you note any high CPU/memory usage that could be causing an issue?

**Comment by JK100001** [comment, marked as answer]
Finally, after a few days of my weekend, I went to the server control panel to look at the status of the streams and they all work.  That is, they recovered themselves. Unfortunately, I can't suggest what helped to solve the problem.

**Reply by tremzo** [reply]
Thats really good that you are now up and running! If you experience the issue again, please let us know.
